packages/semantic-dom-diff/get-diffable-html.js

```diff
@@ -0,0 +1,243 @@
+   * @param {Attr} attr
+   */
+  function getAttributeString(el, attr) {
+    if (attr.name === 'class') {
+      // @ts-ignore
+      const sortedClasses = [...el.classList.values()].sort().join(' ');
+      return ` class="${sortedClasses}"`;
+    }
+    return ` ${attr.name}="${attr.value}"`;
+  }
+
+  /**
+   * @param {Element} el
+   * @param {Attr} attr
+   */
+  function isIgnoredAttribute(el, attr) {
+    if (ignoreAttributes.includes(attr.name)) {
+      return true;
+    }
+
+    return !!ignoreAttributesForTags.find(e => {
+      if (!e.tags || !e.attributes) {
+        throw new Error(
+          `An object entry to ignoreAttributes should contain a 'tags' and an 'attributes' property.`,
+        );
+      }
+      return e.tags.includes(el.nodeName.toLowerCase()) && e.attributes.includes(attr.name);
+    });
+  }
+
+  const sortAttribute = (a, b) => a.name.localeCompare(b.name);
+
+  /** @param {Element} el */
+  function getAttributesString(el) {
+    let attrStr = '';
+    const attributes = Array.from(el.attributes)
+      .filter(attr => !isIgnoredAttribute(el, attr))
+      .sort(sortAttribute);
+
+    if (attributes.length === 1) {
+      attrStr = getAttributeString(el, attributes[0]);
+    } else if (attributes.length > 1) {
+      for (let i = 0; i < attributes.length; i += 1) {
+        attrStr += `\n${getIndentation()} ${getAttributeString(el, attributes[i])}`;
+      }
+      attrStr += `\n${getIndentation()}`;
+    }
+
+    return attrStr;
+  }
+
+  /** @param {Element} el */
+  function printOpenElement(el) {
+    text += `${getIndentation()}<${el.localName}${getAttributesString(el)}>\n`;
+  }
+
+  /** @param {Node} node */
+  function onNodeStart(node) {
+    // don't print this node if we should ignore it
+    if (node.nodeName === 'DIFF-CONTAINER' || ignoreTags.includes(node.nodeName.toLowerCase())) {
+      return;
+    }
+
+    // don't print this node if it was already printed, this happens when
+    // crawling upwards after handling children
+    if (handledNodeStarted.has(node)) {
+      return;
+    }
+    handledNodeStarted.add(node);
+
+    if (node instanceof Text) {
+      printText(node);
+    } else if (node instanceof Element) {
+      printOpenElement(node);
+    } else {
+      throw new Error(`Unknown node type: ${node}`);
+    }
+  }
+
+  /** @param {Element} el */
+  function printCloseElement(el) {
+    if (el.localName === 'diff-container' || VOID_ELEMENTS.includes(el.localName)) {
+      return;
+    }
+
+    text += `${getIndentation()}</${el.localName}>\n`;
+  }
+
+  /** @param {Node} node */
+  function onNodeEnd(node) {
+    // don't print this node if we should ignore it
+    if (ignoreTags.includes(node.nodeName.toLowerCase())) {
+      return;
+    }
+
+    if (node instanceof Element) {
+      printCloseElement(node);
+    }
+  }
+
+  const container = document.createElement('diff-container');
+  container.innerHTML = html;
+
+  const walker = document.createTreeWalker(
```

Comment:
as I did some adjustments for lit-html on this as well I know that IE doesn' t like that

you will need something like this
```
    // Edge needs all 4 parameters present; IE11 needs 3rd parameter to be null
    const walker = document.createTreeWalker(
        element.content,
        133 /* NodeFilter.SHOW_{ELEMENT|COMMENT|TEXT} */,
        null,
        false);
```

Reply:
thanks!